### DISCUSSION TITLE
IPAM deployment using Separate App Services behind App Gateway

### DISCUSSION THREAD

**Discussion by esdcharles** [Azure/ipam#123] [Q&A]
Hey folks,

Not sure if the Q&A is intended for this but I thought I'd reach out to inquire if anyone has implemented this solution using separate App Services (via ACR) behind an App Gateway along with your resources (Cosmos DB, KV etc.) all leveraging Private Endpoints . If so did you encounter any unique challenges that could be shared? Have you incorporated this into any automation? Admittingly I have been struggling to get the IPAM solution going as a completely internal solution and have been doing through the discussions regularly to gather as much as I can from others and their approach. It be nice to hear everyone..

Thanks!

**Comment by DCMattyG** [comment]
Hi @binarybfc, I welcome everyone to chime in here, but allow me to add the view from my purview....

This is probably one of the most requested items I've seen to date. We've considered building some automation (maybe a separate script), but the biggest concern here that we've seen with the already existing ```deploy.ps1``` is that everyone does things *slightly* different, and accounting for every possible corner case is quite difficult.

We've talk about publishing some "alternative architectures", including the one you're describing. If we added a section to the docs which describe a few alternatives (such as AppGateway & AKS), and the high-level requirements of setting up such an environment, would that be helpful? I think this would be the best path as then each customer could take those reference designs and tweak them to meet the unique requirements of their environments.

**Reply by DeadlyDragon** [reply]
@DCMattyG I am looking at doing something similar by having an app service for the engine and an app service for the UI. I noticed there is a react_app variable for specifying engine url, however, when I set this it still seems to talk to itself.

Are there any extra steps? (besides updating the app registration redirect uri's)

**Reply by DCMattyG** [reply]
Hey @DeadlyDragon, you need to set an Application Setting (Environment Variable) on **BOTH** the UI App Service, an the Engine App Service:

UI App Service --> REACT_APP_IPAM_ENGINE_URL: https://engineapp.azurewebsites.net
Engine App Service --> IPAM_UI_URL: https://uiapp.azurewebsites.net

The setting on the UI app changes the URL location for API calls and the setting on the Engine app adds the UI URL into CORS.

Hope that helps, please let me know if you have any further issues!

**Reply by DeadlyDragon** [reply]
@DCMattyG Thanks, need to improve my search and discovery, that works! thanks!
